Sources/SpeziViews/Views/Text/MarkdownView.swift

```diff
@@ -27,67 +27,49 @@ public struct MarkdownView: View {
     public enum Error: LocalizedError {
         case markdownLoadingError
         
+        
         public var errorDescription: String? {
-            "Failed to load the markdown text."
+            LocalizedStringResource("MARKDOWN_LOADING_ERROR", bundle: .atURL(from: .module)).localizedString()
         }
         
         public var recoverySuggestion: String? {
-            "Please check the content of the markdown text."
+            LocalizedStringResource("MARKDOWN_LOADING_ERROR_RECOVERY_SUGGESTION", bundle: .atURL(from: .module)).localizedString()
         }
 
         public var failureReason: String? {
-            "The system wasn't able to parse the given markdown text, indicating an invalid markdown text."
+            LocalizedStringResource("MARKDOWN_LOADING_ERROR_FAILURE_REASON", bundle: .atURL(from: .module)).localizedString()
         }
     }
     
     
     private let asyncMarkdown: () async -> Data
     
-    @State private var markdown: Data?
+    @State private var markdownString: AttributedString?
     @Binding private var state: ViewState
     
     
-    private var markdownString: AttributedString {
-        guard let markdown,
-              let markdownString = try? AttributedString(
-                markdown: markdown,
-                options: .init(interpretedSyntax: .inlineOnlyPreservingWhitespace)
-              ) else {
-            Task { @MainActor in
-                state = .error(Error.markdownLoadingError)
-            }
-            return AttributedString(
-                String(localized: "MARKDOWN_LOADING_ERROR", bundle: .module)
-            )
-        }
-        
-        Task { @MainActor in
-            state = .idle
-        }
-        return markdownString
-    }
-    
-    
     public var body: some View {
         VStack {
-            if markdown == nil {
-                ProgressView()
-                    .padding()
-            } else {
+            if let markdownString {
                 Text(markdownString)
+            } else {
+                ProgressView()
                     .padding()
             }
         }
             .task {
                 state = .processing
-                markdown = await asyncMarkdown()
+                markdownString = parse(
+                    markdown: await asyncMarkdown()
+                )
+                state = .idle
```

Comment:
This statement overrides the `.error` state in `parse(markdown:)`. We should probably move all the `state` management in the parse function and then only set `.idle` in the non-guard state.

Reply:
Great catch, shouldn't quickly incorporate your feedback in London Heathrow 😀